packages/shared/src/observe-slot-text.ts

```diff
@@ -39,7 +37,25 @@ export function ObserveSlotText<T extends Constructor<ReactiveElement>>(
         extends constructor
         implements SlotTextObservingInterface
     {
-        private [slotElementObserver]!: MutationObserver;
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        constructor(...args: any[]) {
+            super(args);
+
+            new MutationController(this, {
+                config: {
+                    characterData: true,
+                    subtree: true,
+                },
+                callback: (mutationsList: Array<MutationRecord>): boolean => {
+                    for (const mutation of mutationsList) {
+                        if (mutation.type === 'characterData') {
+                            this.manageTextObservedSlot();
```

Comment:
Would it make sense to early return here, so `this.manageTextObservedSlot()` is guaranteed to only be called once?

```suggestion
                            this.manageTextObservedSlot();
                            return true;
```